File: app/src/main/java/com/cristhian/gametobeat/database/GameViewModel.java
package com.cristhian.gametobeat.database;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.cristhian.gametobeat.util.FirebaseUtil;
import com.google.firebase.analytics.FirebaseAnalytics;

import java.util.List;

public class GameViewModel extends AndroidViewModel {

    private GameRepository mRepository;

    private LiveData<List<Game>> mGamesBeat;
    private LiveData<List<Game>> mGamesPlaying;
    private LiveData<List<Game>> mGamesWant;

    private FirebaseAnalytics mFirebaseAnalytics;

    public GameViewModel(@NonNull Application application) {
        super(application);
        mRepository = new GameRepository(application);
        mGamesBeat = mRepository.getGamesBeat();
        mGamesPlaying = mRepository.getGamesPlaying();
        mGamesWant = mRepository.getGamesWant();
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(application);
    }

    public LiveData<List<Game>> getGamesPlaying() {
        return mGamesPlaying;
    }

    public LiveData<List<Game>> getGamesBeat() {
        return mGamesBeat;
    }

    public LiveData<List<Game>> getGamesWant() {
        return mGamesWant;
    }

    public void insert(Game game) {
        mRepository.insert(game);
        FirebaseUtil.logGameEvent(mFirebaseAnalytics, game.getName(), FirebaseUtil.PARAM_GAME_ADD);
    }

    public void update(Game game) {
        mRepository.update(game);
        FirebaseUtil.logGameEvent(mFirebaseAnalytics, game.getName(), FirebaseUtil.PARAM_GAME_UPDDATE);
    }

    public void delete(Game game) {
        mRepository.delete(game);
        FirebaseUtil.logGameEvent(mFirebaseAnalytics, game.getName(), FirebaseUtil.PARAM_GAME_REMOVE);
    }

    public LiveData<Game> loadGameById(Game game) {
        return mRepository.loadGameById(game);
    }
}

File: app/src/main/java/com/cristhian/gametobeat/database/GameViewModel.kt
package com.cristhian.gametobeat.database

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.cristhian.gametobeat.util.FirebaseUtil
import com.cristhian.gametobeat.util.FirebaseUtil.logGameEvent
import com.google.firebase.analytics.FirebaseAnalytics

class GameViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: GameRepository = GameRepository(application)

    val gamesBeat: LiveData<List<Game?>?>?
    val gamesPlaying: LiveData<List<Game?>?>?
    val gamesWant: LiveData<List<Game?>?>?
    private val mFirebaseAnalytics: FirebaseAnalytics

    fun insert(game: Game) {
        mRepository.insert(game)
        logGameEvent(mFirebaseAnalytics, game.name, FirebaseUtil.PARAM_GAME_ADD)
    }

    fun update(game: Game) {
        mRepository.update(game)
        logGameEvent(mFirebaseAnalytics, game.name, FirebaseUtil.PARAM_GAME_UPDDATE)
    }

    fun delete(game: Game) {
        mRepository.delete(game)
        logGameEvent(mFirebaseAnalytics, game.name, FirebaseUtil.PARAM_GAME_REMOVE)
    }

    fun loadGameById(game: Game?): LiveData<Game?>? {
        return mRepository.loadGameById(game!!)
    }

    init {
        gamesBeat = mRepository.gamesBeat
        gamesPlaying = mRepository.gamesPlaying
        gamesWant = mRepository.gamesWant
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(application)
    }
}